Source/LibationFileManager/Templates.cs

```diff
@@ -247,6 +252,89 @@ private static List<List<string>> GetPathParts(IEnumerable<string> templateParts
+		/// <summary> Max must have a 1 or 2-digit number </summary>
+		[GeneratedRegex(@"[Mm]ax\(\s*?(\d{1,2})\s*?\)")]
+		private static partial Regex NamesMaxRegex();
+
+		private static string NameListFormatter(ITemplateTag templateTag, IEnumerable<string> names, string formatString)
+		{
+			var humanNames = names.Select(n => new HumanName(removeSuffix(n), Prefer.FirstOverPrefix));
+
+			var sortedNames = sort(humanNames, formatString);
+			var nameFormatString = format(formatString, defaultValue: "{T} {F} {M} {L} {S}");
+			var separatorString = separator(formatString, defaultValue: ", ");
+			var maxNames = max(formatString, defaultValue: humanNames.Count());
+
+			var formattedNames = string.Join(separatorString, sortedNames.Take(maxNames).Select(n => formatName(n, nameFormatString)));
+
+			while (formattedNames.Contains("  "))
+				formattedNames = formattedNames.Replace("  ", " ");
+
+			return formattedNames;
+
+			static string removeSuffix(string namesString)
+			{
+				namesString = namesString.Replace('’', '\'').Replace(" - Ret.", ", Ret.");
+				int dashIndex = namesString.IndexOf(" - ");
+				return (dashIndex > 0 ? namesString[..dashIndex] : namesString).Trim();
+			}
+
+			static IEnumerable<HumanName> sort(IEnumerable<HumanName> humanNames, string formatString)
+			{
+				var sortMatch = NamesSortRegex().Match(formatString);
+				return
+					sortMatch.Success
+					? sortMatch.Groups[1].Value == "F" ? humanNames.OrderBy(n => n.First)
+						: sortMatch.Groups[1].Value == "M" ? humanNames.OrderBy(n => n.Middle)
+						: sortMatch.Groups[1].Value == "L" ? humanNames.OrderBy(n => n.Last)
+						: humanNames
+					: humanNames;
+			}
+
+			static string format(string formatString, string defaultValue)
+			{
+				var formatMatch = NamesFormatRegex().Match(formatString);
+				return formatMatch.Success ? formatMatch.Groups[1].Value : defaultValue;
+			}
+
+			static string separator(string formatString, string defaultValue)
+			{
+				var separatorMatch = NamesSeparatorRegex().Match(formatString);
+				return separatorMatch.Success ? separatorMatch.Groups[1].Value : defaultValue;
+			}
+
+			static int max(string formatString, int defaultValue)
+			{
+				var maxMatch = NamesMaxRegex().Match(formatString);
+				return maxMatch.Success && int.TryParse(maxMatch.Groups[1].Value, out var max) ? int.Max(1, max) : defaultValue;
+			}
+
+			static string formatName(HumanName humanName, string nameFormatString)
+			{
+				//Single-word names parse as first names. Use it as last name.
+				var lastName = string.IsNullOrWhiteSpace(humanName.Last) ? humanName.First : humanName.Last;
+
+				nameFormatString
+					= nameFormatString
+					.Replace("{T}", "{0}")
+					.Replace("{F}", "{1}")
+					.Replace("{M}", "{2}")
+					.Replace("{L}", "{3}")
+					.Replace("{S}", "{4}");
+
+				return string.Format(nameFormatString, humanName.Title, humanName.First, humanName.Middle, lastName, humanName.Suffix).Trim();
+			}
```

Comment:
I don't think we ever discussed local functions or how you feel about them. I like them because they make methods more readable while maintaining encapsulation, but I don't think I've ever seen you use them. What are your thoughts?

Reply:
I love them in moderation. They're something I've long wished for without knowing what to call them. If there are more than a few, it might be an indication that it's time for a new class. That's a case-by-case judgement call.